src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java

```diff
@@ -17,47 +17,102 @@
 
 import io.github.anthorx.parquet.sql.record.Record;
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
 import org.apache.parquet.hadoop.ParquetFileReader;
 import org.apache.parquet.hadoop.ParquetReader;
 import org.apache.parquet.hadoop.api.ReadSupport;
 import org.apache.parquet.hadoop.util.HadoopInputFile;
+import org.apache.parquet.hadoop.util.HiddenFileFilter;
 import org.apache.parquet.io.InputFile;
 import org.apache.parquet.schema.MessageType;
 import org.apache.parquet.schema.Type;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public class SQLParquetReaderWrapper {
 
-  private final ParquetReader<Record> parquetReader;
-  private final MessageType schema;
+  private ParquetReader<Record> currentParquetReader;
+  private final Iterator<ParquetReader<Record>> parquetReaderIterator;
+  private MessageType schema;
 
-  public SQLParquetReaderWrapper(String filePath) throws IOException {
-    InputFile inputFile = HadoopInputFile.fromPath(new Path(filePath), new Configuration());
+  public SQLParquetReaderWrapper(String stringFilePath) throws IOException {
+    Path filePath = new Path(stringFilePath);
+    Configuration configuration = new Configuration();
```

Comment:
To work in different contexts, clients should be able to provide their own hadoop conf. 
I suggest adding a constructor with Configuration in parameter.

Reply:
True. Change incoming!